tests/test_class.py

```diff
@@ -40,3 +40,43 @@ def set_attr(self):
 
 
 
+class SlotsBase(object):
+    __slots__ = []
+    def __eq__(self, other):
+        ans = self.__class__ == other.__class__
+        slots = self.__slots__
+        if isinstance(slots,str):
+            slots = [slots]
+        return ans and all(getattr(self,i)==getattr(other,i) for i in self.__slots__)
+    
```

Comment:
The mistake is actually in this line where I should have used the new variable slots instead of self.__slots__ . This is because __slots__ can either be a string or a list of strings.